src/Middleware.php

```diff
@@ -91,12 +91,13 @@ public function __invoke(Request $request, Response $response, callable $out = n
      *
      * A handler can be any callable, or an object with a handle() method.
      *
-     * Handlers with arity >= 4 are considered error handlers, and will
+     * Handlers with parity >= 4 are considered error handlers, and will
      * be executed when a handler calls $next with an argument or raises
      * an exception.
      *
      * @param string|callable|object $path Either a URI path prefix, or a handler
      * @param null|callable|object $handler A handler
+     * @return $this|void
```

Comment:
We can remove `void` as a return, and rename `$this` to `self` (per phpdoc guidelines).